sbn/codingconv/CodingConventions.md

```diff
@@ -503,9 +503,9 @@ It should be used judiciously (and sparsely).
   [user-defined literal operators](https://en.cppreference.com/w/cpp/language/user_literal)
   (e.g. `using namespace std::string_literals`);
 
-**[F]** `using namespace` directives are nevertheless always thoroughly
+**[F]** `using` directives are nevertheless always thoroughly
   **forbidden** in global scope of header files.
-  Instead, importing specific symbols at local scope should be preferred
+  You may import specific symbols at local scope (i.e. within the body of your `class`) if absolutely necessary for readability
```

Comment:
I'm not sure I understand the example. Does `return std::empty(coll) ? typename Coll::value_type{}: *std::begin(coll);` not work?

Reply:
If you are using a `Coll` that does not have a `begin()` method (e.g. because it's a C data structure or as it was in the past `TCollection` from ROOT, or maybe it's like `std::filesystem::directory_iterator`), the portal to STL is to have a free function `begin(Coll const&)` (and friends), ideally in the same namespace as `Coll` (alternatively, at global scope). So, we may implement `begin(linked_list*)` and `begin(TCollection const&)` ourselves and welcome there in STL world.
`std::begin()` would not find that function, but it covers the cases where that function is not available.